channels/applicationinsights-channel-js/src/SendBuffer.ts

```diff
@@ -185,36 +185,39 @@ export class ArraySendBuffer extends BaseSendBuffer implements ISendBuffer {
     }
 }
 
+type StorageGetter = typeof utlGetSessionStorage;
+type StorageSetter = typeof utlSetSessionStorage;
+
 /*
- * Session storage buffer holds a copy of all unsent items in the browser session storage.
+ * Base storage buffer for simple sync-based string storage
  */
-export class SessionStorageSendBuffer extends BaseSendBuffer implements ISendBuffer {
+export class StringStorageSendBuffer extends BaseSendBuffer implements ISendBuffer {
     static BUFFER_KEY = "AI_buffer";
     static SENT_BUFFER_KEY = "AI_sentBuffer";
 
     // Maximum number of payloads stored in the buffer. If the buffer is full, new elements will be dropped.
     static MAX_BUFFER_SIZE = 2000;
 
-    constructor(logger: IDiagnosticLogger, config: ISenderConfig) {
+    constructor(logger: IDiagnosticLogger, config: ISenderConfig, setStorage: StorageSetter, getStorage: StorageGetter) {
```

Comment:
To support v3 and dynamic configuration support, I believe we change these classes (or at least their usage) quite a bit.

So If we could keep the name the same it would simplify the cherry-pick process at this point

Reply:
Done. I've leveraged the `SessionStorageSendBuffer` as a base class for `LocalStorageSendBuffer`, but tried to keep the impacts minimal there, so hopefully it's trivial to adjust to the new approaches.